kratos/includes/initial_state.h

```diff
@@ -0,0 +1,279 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//
+//
+
+#if !defined(KRATOS_INITIAL_STATE_H_INCLUDED)
+#define  KRATOS_INITIAL_STATE_H_INCLUDED
+
+// System includes
+#include <iostream>
+#include <string>
+
+// External includes
+
+// Project includes
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class InitialState
+ * @ingroup StructuralMechanicsApplication
+ * @brief Define the initial state of the material in terms of initial stress/strain/F
+ * @details Storages the information regarding initial stresses/strains/F
+ * @author Alejandro Cornejo
+ */
+class KRATOS_API(KRATOS_CORE) InitialState
+{
+    public:
+        ///@name Type Definitions
+        ///@{
+        typedef std::size_t SizeType;
+
+        enum class InitialImposingType {StrainOnly = 0, StressOnly = 1, DeformationGradientOnly = 2, StrainAndStress = 3, DeformationGradientAndStress = 4};
```

Comment:
this is an enum, so in kratos we normally use it all capitals, something like STRAIN_ONLY, STRESS_ONLY, DEFORMATION_GRADIENT_ONLY ...

Reply:
ok